yt_dlp/extractor/redgifs.py

```diff
@@ -0,0 +1,113 @@
+# coding: utf-8
+
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+from ..utils import (
+    ExtractorError,
+    int_or_none,
+    qualities,
+)
+
+
+class RedGifsIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(?:www|thumbs2?)\.)?redgifs\.com/(?:watch/)?(?P<id>[^-/?#\.]+)'
+    _TESTS = [{
+        'url': 'https://www.redgifs.com/watch/squeakyhelplesswisent',
+        'info_dict': {
+            'id': 'squeakyhelplesswisent',
+            'ext': 'mp4',
+            'title': 'Hotwife Legs Thick',
+            'timestamp': 1636287915,
+            'upload_date': '20211107',
+            'uploader': 'ignored52',
+            'duration': 16,
+            'view_count': int,
+            'like_count': int,
+            'categories': list,
+            'age_limit': 18,
+        }
+    }, {
+        'url': 'https://thumbs2.redgifs.com/SqueakyHelplessWisent-mobile.mp4#t=0',
+        'info_dict': {
+            'id': 'squeakyhelplesswisent',
+            'ext': 'mp4',
+            'title': 'Hotwife Legs Thick',
+            'timestamp': 1636287915,
+            'upload_date': '20211107',
+            'uploader': 'ignored52',
+            'duration': 16,
+            'view_count': int,
+            'like_count': int,
+            'categories': list,
+            'age_limit': 18,
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url).lower()
+
+        video_info = self._download_json(
+            'https://api.redgifs.com/v2/gifs/%s' % video_id,
+            video_id, 'Downloading video info')
+        if 'error' in video_info:
+            raise ExtractorError('RedGifs said: ' + video_info['error'], expected=True)
+
+        gif = video_info['gif']
+        urls = gif['urls']
+
+        # redgifs do not have title
+        title = ' '.join(gif.get('tags')) or 'RedGifs'
+        timestamp = int_or_none(gif.get('createDate'))
+        uploader = gif.get('userName')
+        view_count = int_or_none(gif.get('views'))
+        like_count = int_or_none(gif.get('likes'))
+        age_limit = 18
+
+        orig_width = int_or_none(gif.get('width'))
+        orig_height = int_or_none(gif.get('height'))
+
+        duration = int_or_none(gif.get('duration'))
+
+        categories = gif.get('tags') or []
```

Comment:
put all vars that are actually never used directly in the return statement

Reply:
Done.